src/utils/loginUtils.js

```diff
@@ -4,88 +4,156 @@ import facebookKey from '../../config/facebook-app-key';
 export function verifyToken(token, callback) {
-  fetch('http://localhost:5000/auth/' + token).then(response => { // change url
+  fetch('http://127.0.0.1:5000/auth/' + token).then(response => { // change url
     var isValid = JSON.parse(response._bodyInit)
+    console.log('$$$$$$', response._bodyInit)
     if (!isValid) {
-      loginUser(); //if token is expired or inValid redirect to login
+      loginUser(callback); //if token is expired or inValid redirect to login
     } else if (response.isExpired) { // Currently not handling difference between invalid and expired token
-      loginUser();
+      loginUser(callback);
     } else if (isValid) {
-      callback(); //if token is valid app opens
+      getProfile(callback)
+      // callback(); //if token is valid app opens
     }
+  }).catch(err => {
+    console.log('loginUtils -> verifyToken -> error', err)
+    loginUser(callback)
   })
 }
 
 
-// function to see if user info exists in AsyncStorage
-export function checkStorage(callback) {
+// // function to see if user info exists in AsyncStorage
+// export function checkStorage(callback) {
+//   AsyncStorage.getItem('token', (err, token) => {
+//     if (err) {
+//       console.log('loginUtils -> checkStorage -> error: ', err);
+//       loginUser(callback);
+//     } else {
+//       token = JSON.parse(token);
+//       verifyToken(token, callback);
+//     }
+//   })
+// }
+
+
+// export function logOutUser(callback) {
+//   var items = ['token', 'profile'];
+//   AsyncStorage.multiRemove(items, (err) => {
+//     if (err) {
+//       console.log('loginUtils -> logOutUser -> remove token', err);
+//     } else {
+//       console.log('Profile Removed');
+//       loginUser(callback);
+//     }
+//   });
+// }
+
+
+export function authorizeUser(callback) {
   AsyncStorage.getItem('token', (err, token) => {
     if (err) {
-      console.log('loginUtils -> checkStorage -> error: ', err);
-      loginUser();
+      console.log('loginUtils -> authorizeUser -> error: ', err);
+      loginUser(callback);
     } else {
       token = JSON.parse(token);
       verifyToken(token, callback);
     }
   })
 }
 
-
-export function logOutUser() {
-  var items = ['token', 'profile'];
-  AsyncStorage.multiRemove(items, (err) => {
+export function loginUser(callback) {
+  var lock = new Auth0Lock(facebookKey);
+  lock.show({}, (err, profile, token) => {
     if (err) {
-      console.log('loginUtils -> logOutUser -> remove token', err);
+      console.log('loginUtils -> loginUser -> Error', err);
     } else {
-      console.log('Profile Removed');
-      loginUser();
+      saveProfileAndToken(profile, token, callback);
     }
   });
-}
\ No newline at end of file
+}
+
+export function saveProfileAndToken(profile, token, callback) {
+  AsyncStorage.setItem('profile', JSON.stringify(profile), (err, success) => {
+    if (err) {
+      console.log('loginUtils -> saveProfileAndToken -> profile', err);
```

Comment:
As is, the callback will not be called if we get an error. Should probably call it here with callback(err, null).